lib/extensions/postgres_cdc_rls/subscription_manager.ex

```diff
@@ -169,6 +176,26 @@ defmodule Extensions.PostgresCdcRls.SubscriptionManager do
     {:noreply, %{state | delete_queue: %{ref: ref, queue: q1}}}
   end
 
+  def handle_info(:check_no_users, %{subscribers_tid: tid, no_users_ts: ts} = state) do
+    H.cancel_timer(state.no_users_ref)
+
+    ts2 =
+      case {:ets.info(tid, :size), ts != nil && ts + @stop_after < now()} do
+        {0, true} ->
+          Logger.info("Stop tenant #{state.id} because of no connected users")
+          Rls.handle_stop(state.id, 15_000)
```

Comment:
@abc3 let's say this synchronous action does take 15 seconds can't this block and choke the process if the tenant suddenly has an influx of subscribers sending `:subscribed` messages? won't this potentially prevent the dynamic supervisor tree from shutting down?

Reply:
If some users have tried to subscribe, they will receive the `postgres_cdc_down` when `handle_stop/2` is terminated and will try to resubscribe and make a new db connection.

The `handle_stop/2` function will stop the existing supervisor `PID` (or crush `SubscriptionManager` because timeout raises an error).